functions/Parse-DbaCsvMetadata.ps1

```diff
@@ -0,0 +1,360 @@
+                #  [single]    Single-precision 32-bit floating point number
+                #  [double]    Double-precision 64-bit floating point number
+                #  [DateTime]  Date and Time
+                #  [string]    Fixed-length string of Unicode characters
+                
+                $hash = [pscustomobject]@{
+                    NumberedHeader = $HeaderNumber    
+                    NamedHeader = $HeaderName
+                    SampleValue = $RowColumns[$_]
+                    SampleLength = $RowColumns[$_].Length
+                    char = ([char]::TryParse($RowColumns[$_], [ref]0))
+                    byte = ([byte]::TryParse($RowColumns[$_], [ref]0))
+                    int16 = ([int16]::TryParse($RowColumns[$_], [ref]0))
+                    int32 = ([int32]::TryParse($RowColumns[$_], [ref]0))
+                    long = ([long]::TryParse($RowColumns[$_], [ref]0))
+                    bool = ([bool]::TryParse($RowColumns[$_], [ref]0))
+                    decimal = ([decimal]::TryParse($RowColumns[$_], [ref]0))
+                    single = ([single]::TryParse($RowColumns[$_], [ref]0))
+                    double = ([double]::TryParse($RowColumns[$_], [ref]0))
+                    DateTime = ([DateTime]::TryParse($RowColumns[$_], [ref]0))
+                    string = $true
+                }
+                $results += $hash
+            } `
+            -end { Write-Verbose "[csv file] Combining metadata." }
+            $hash = [pscustomobject]@{    
+                FileName = $FileName
+                HasFieldsEnclosedInQuotes = $HasFieldsEnclosedInQuotes
+                Properties = $results
+            }
+            $MetaDataCollection += $hash
+        }
+        
+        if($MetaDataCollection.Count -gt 1) 
+        {
+            
+            Write-Verbose "[csv file] Comparing the column length of the csv files."
+            for ($i = 0; $i -lt $MetaDataCollection.Count -1; $i++) 
+            {
+                $current = $MetaDataCollection[$i]
+                $next = $MetaDataCollection[$i+1]
+
+                $msg = ("[csv file] Match {0} [{1}], {2} [{3}]" -f $current.FileName, $current.Properties.length, $next.FileName , $next.Properties.length)
+                if ($current.Properties.length -eq $next.Properties.length) 
+                {
+                    Write-Verbose $msg
+                }
+                else 
+                {
+                    Write-Warning $msg.Replace("Match","Difference between")
+                    Get-UserPrompt
+                }
+            }
+        }
+        $sw.Stop()
+        Write-Verbose ("[stopwatch] All process complete. Elapsed: {0}" -f $sw.Elapsed)
+        
+    }
+
+    END 
+    {
+        if ($abort)
+        {
+            Write-Verbose "[csv file] No CSV file selected, aborted."
+            return
+        }
+        return $MetaDataCollection
+    }
+}
+
+# This is a test output

```

Comment:
Please move or remove. Comments should be either in the top block or in the comments of the PR.